gobblin-utility/src/main/java/gobblin/util/AvroUtils.java

```diff
@@ -98,19 +99,31 @@
 
   /**
    * Helper method that does the actual work for {@link #getFieldSchema(Schema, String)}
-   * @param schema passed from {@link #getFieldValue(Schema, String)}
-   * @param pathList passed from {@link #getFieldValue(Schema, String)}
+   * @param schema passed from {@link #getFieldSchema(Schema, String)}
+   * @param pathList passed from {@link #getFieldSchema(Schema, String)}
    * @param field keeps track of the index used to access the list pathList
    * @return the schema of the field
    */
   private static Optional<Schema> getFieldSchemaHelper(Schema schema, List<String> pathList, int field) {
-    if (schema.getField(pathList.get(field)) == null) {
+    if (schema.getType() == Type.RECORD && schema.getField(pathList.get(field)) == null) {
       return Optional.absent();
     }
-    if ((field + 1) == pathList.size()) {
-      return Optional.fromNullable(schema.getField(pathList.get(field)).schema());
+    switch (schema.getType()) {
+      case UNION:
+        throw new AvroRuntimeException("Union of complex types cannot be handled : " + schema);
+      case MAP:
```

Comment:
Can you please include a couple of test cases: 
1. That retrieves a schema that has map in its path; and 
2. That goes the happy path of nested records (as was the only supported case) since there isn't already a test case for it